drivers: wifi: Add support to override FW load

krish2718 opened the pull request:
For using external flash the tooling is only available in NCS, so, use the existing Kconfig option to override the FW loading.

krish2718 committed:
drivers: wifi: Add support to override FW load

For using external flash the tooling is only available in NCS, so, use
the existing Kconfig option to override the FW loading.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

krish2718 committed:
drivers: wifi: nrf: Remove unused linker script

This script is needed only when external flash is used to store nRF70
firmware patches. This also uses PM which is supported only in NCS.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

jukkar commented on drivers/wifi/nrfwifi/Kconfig.nrfwifi:
```diff
@@ -130,15 +130,12 @@ config NRF_WIFI_IF_AUTO_START
 config NRF_WIFI_PATCHES_BUILTIN
 	bool "Store nRF70 FW patches as part of the driver"
-	default y
+	default y if !NRF_WIFI_BUILD_ONLY_MODE
```
I think you can say here
`default y`
as the "depends on ..." will prevent the option to be enabled.

nordicjm reviewed (dismissed):
Jukkar's point is valid

krish2718 committed:
drivers: nrfwifi: Fix rebuilding when FW blobs are changes

Whenever FW blobs are updated manually, we need to tell cmake to
auto-build the source files to use the latest firmware without doing a
pristine build.

This adds a custom target to be run with nRF Wi-Fi driver and updates
timestamp of fw_load.c to rebuild whenevr the blob is updated.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

nordicjm commented on drivers/wifi/nrfwifi/CMakeLists.txt:
```diff
@@ -169,6 +169,22 @@ endif()
 zephyr_compile_definitions(
   -DCONFIG_NRF_WIFI_FW_BIN=${NRF70_PATCH}
 )
+
+# Rebuild fw_load.c whenever the firmware binary changes
+# a bit crude way to do it, but it works
+add_custom_command(
+  OUTPUT ${CMAKE_SOURCE_DIR}/src/fw_load.c
+  COMMAND ${CMAKE_COMMAND} -E touch ${CMAKE_SOURCE_DIR}/src/fw_load.c
+  DEPENDS ${NRF70_PATCH}
+  COMMENT "Checking firmware blobs ${NRF70_PATCH}"
+)
+
+add_custom_target(
+  check_firmware_blobs ALL
+  DEPENDS ${CMAKE_SOURCE_DIR}/src/fw_load.c
+)
+
+add_dependencies(nrfwifi check_firmware_blobs)
```
this and all above should be indented by 2 spaces since inside of if()

krish2718 committed:
drivers: nrfwifi: Fix the indentation

The Cmake indentation is two spaces.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>
